src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -532,12 +532,16 @@ public void onThrowable(final Throwable t) {
         public T onCompleted(final Response response) throws Exception {
             if (isSuccess(response)) {
                 final T value = MAPPER.readValue(response.getResponseBody(), clazz);
```

Comment:
In Java,`T` is a [`generic`](https://docs.oracle.com/javase/tutorial/java/generics/types.html) which means the value of the class is deffered at compile time and determined during runtime. When we are calling the function, we set what class `T` will be in this particular invocation of the function. This can be see in AsyncLobClient when we call the `execute` function and pass in `PostcardResponse.class` (or other classes) as an argument to be set as `clazz` (class is a restricted keyword) which is then passed again to the `GuavaFutureConverter`. `AbstractResponse` is an [`abstract class`](https://docs.oracle.com/javase/tutorial/java/IandI/abstract.html) which means that this class can't be instantiated but subclasses. Every possible response inherits this class which means that the response has the properties declared in `AbstractResponse`. Because `value` is of class `T` which hasn't been declared yet, I can't simply call `setHeaders()` and that is why I have to cast it in order to call that function. Because we know that all possible values of `T` inherit from `AbstractResponse`, it is safe to cast it to that class in order to call that function.